client/apps/plugin-status/state/actions.js

```diff
@@ -32,16 +36,57 @@ export const save = () => ( dispatch, getState ) => {
 	};
 
 	api.post( api.url.selfHelp(), data )
-		.then( () => dispatch( NoticeActions.successNotice( translate( 'Your changes have been saved.' ), {
-			duration: 5000,
-		} ) ) )
+		.then( () => {
+			dispatch( NoticeActions.successNotice( translate( 'Your changes have been saved.' ), {
+				duration: 5000,
+			} ) );
+
+			if ( state.debug_saving ) {
+				dispatch( setDebug( state.debug_enabled, false ) );
+			}
+			if ( state.logging_saving ) {
+				dispatch( setLogging( state.logging_enabled, false ) );
+			}
+		} )
 		.catch( ( error ) => {
 			if ( _.isString( error ) ) {
 				dispatch( NoticeActions.errorNotice( error ) );
+			} else if ( _.isObject( error ) ) {
+				dispatch( NoticeActions.errorNotice( translate( 'There was a problem when saving your preferences. Please try again.' ) ) );
 			}
 
-			if ( _.isObject( error ) ) {
-				dispatch( NoticeActions.errorNotice( translate( 'There was a problem when saving your preferences. Please try again.' ) ) );
+			if ( state.debug_saving ) {
+				dispatch( setDebug( ! state.debug_enabled, false ) );
 			}
+			if ( state.logging_saving ) {
+				dispatch( setLogging( ! state.logging_enabled, false ) );
+			}
+		} );
+};
+
+export const checkRestApi = () => ( dispatch, getState ) => {
+	dispatch( { type: PLUGIN_STATUS_REST_REQUEST } );
+
+	const state = getState().status;
+
+	const data = {
+		wcc_debug_on: state.debug_enabled,
+		wcc_logging_on: state.logging_enabled,
+	};
+
+	api.post( api.url.selfHelp(), data )
+		.then( () => {
+			dispatch( {
+				type: PLUGIN_STATUS_REST_RESPONSE,
+				success: true,
+				message: translate( 'Can manage settings and print shipping labels' ),
```

Comment:
Seeing how we're only testing that the debug/logging saving here - it seems like a stretch to claim that settings and labels have been checked.

While it's better than no test, what do you think about changing this to be a bit more generic? Something like "WooCommerce REST API is working properly"?

Reply:
Changed in https://github.com/Automattic/woocommerce-services/pull/1369/commits/85f931d70106d1209b632538f90211055d9b8bfd to "WooCommerce REST API is working properly" vs "Encountered an issue with the WooCommerce REST API".